### PR TITLE
Fix `Object::notification` order (cpp-bindings)

### PR DESCRIPTION
**Pull request by Sauermann**
For the notification-order to work correctly, it is necessary to allow the `p_reversed` argument to be used within cpp. This PR changes the necessary bindings.

- cpp-bindings for https://github.com/godotengine/godot/pull/78634

This is my first cpp-api change, so please review carefully.

### CODE REVIEW

File: include/godot_cpp/classes/wrapped.hpp
Comment (dsnopek) [inline include/godot_cpp/classes/wrapped.hpp:58]: I don't think we need to add `_notification2()`. In godot-cpp, we _can_ break compatibility, and we want to have the same API (as much as is reasonable) as in a Godot module. So, in my opinion, we should just change `_notification()` to have the new second argument.

File: include/godot_cpp/core/class_db.hpp
Comment (dsnopek) [inline include/godot_cpp/core/class_db.hpp:186]: We shouldn't be passing both of these. Since godot-cpp is being updated with awareness of `classdb_register_extension_class2`, it should just pass `nullptr` for the old function pointer.

**Review by dsnopek — CHANGES_REQUESTED** [review]
Thanks!

**Comment by dsnopek** [comment]
Looking great! I manually tested together with the latest Godot PR, and everything seems to be working fine. :-) I'd be happy to approve this after the Godot change is merged, and CI is passing here (the automated tests pass locally for me).

**Comment by Sauermann** [comment]
With godotengine/godot#78634 merged in the main repository, now all checks have passed.

**Review by dsnopek — APPROVED** [review]
Looks great, thanks!

**Pull request merged**
